create user interface

EcLiPsE-code opened the pull request.

EcLiPsE-code committed:
create user interface

EcLiPsE-code committed:
update user interface

EcLiPsE-code committed:
Update .gitignore

EcLiPsE-code committed:
update user interfaces and create store (redux) for system

EcLiPsE-code committed:
update user interface and create sign in dialog

EcLiPsE-code committed:
update navigationPanel and added and update forms signIn and signUp

EcLiPsE-code committed:
update interfaces

EcLiPsE-code committed:
update

EcLiPsE-code committed:
update user interface

EcLiPsE-code committed:
update users interfaces

+ add new screens

default-INT commented on stand-ui/src/components/UI/menu/menu.js:
```diff
@@ -0,0 +1,63 @@
+    let menuItem = []
+
+    useEffect(() => {
+        let {id, menuItem} = props
```
Зачем в useEffect доставать свойства из props и зачем объявлять через let, если ссылка в последствии не меняется?

default-INT commented on stand-ui/src/components/UI/select/userSelect.module.css:
```diff
@@ -0,0 +1,5 @@
+.Select{
```
класс стиля с БОЛЬШОЙ БУКВЫ????

default-INT commented on stand-ui/src/components/UI/table/protocols/protocolsTable.js:
```diff
@@ -0,0 +1,90 @@
+
+const columns = [
+    {id: 'program', label: 'Программа', align: 'center'},
+    {id: 'testType', label: 'Тип испытания', align: 'center'},
```
Зачем юзать React.useState, если ты импортировал эту функцию отдельно?

default-INT commented on stand-ui/src/components/header/header.module.css:
```diff
@@ -0,0 +1,13 @@
+.Header{
```
Что эта за такая мода???? Какой у нас конвеншен??

default-INT commented on stand-ui/src/components/pages/charts/components/chart.js:
```diff
@@ -0,0 +1,73 @@
+
+    const [arrX, setArrX] = useState([])
+
+    const [state, setState] = useState({
```
Что это такое??? Что за состояние???
Большие объекты как состояние обычно не создаются, для этого используется useReducer.
хардкод

default-INT commented on stand-ui/src/containers/navigationPanel/navigationPanel.js:
```diff
@@ -0,0 +1,64 @@
+const NavigationPanel = props => {
+
+    const [anchorEl, setAnchorEl] = React.useState(null);
+    const [state, setState] = useState({
```
снова

default-INT commented on stand-ui/src/containers/navigationPanel/navigationPanel.js:
```diff
@@ -0,0 +1,64 @@
+    };
+
+    const changeDrawer = () => {
+        setState(
```
если ты решил развернуть предыдущий локальный State, используй setState(prevState => {...prevState}), для получения последнего актуального состояния

default-INT commented on stand-ui/src/containers/pages/charts/charts.js:
```diff
@@ -0,0 +1,30 @@
+        <div className={classes.ChartsWrapper}>
+            <div className={classes.Chart}>
+                <UserChart
+                    id={'chart-1'}
```
id='chart-1' - написать не Варик?

default-INT commented on stand-ui/src/containers/pages/personalArea/personalArea.js:
```diff
@@ -0,0 +1,131 @@
+        booleanLabel: false,
+        dataLabel: 'Евгений'
+    })
+    const [surname, setSurname] = useState({
```
приятно конечно, но вместо создания 10 локальный состояний, рациональнее будет использовать useReducer

default-INT commented on stand-ui/src/helerps/randomizer/randomizer.js:
```diff
@@ -0,0 +1,4 @@
+export default function randomInteger(min, max){
+    let rand = min - 0.5 + Math.random() * (max - min + 1);
+    return Math.round(rand);
```
можно просто return сделать

default-INT commented on stand-ui/src/reportWebVitals.js:
```diff
@@ -0,0 +1,13 @@
+const reportWebVitals = onPerfEntry => {
+  if (onPerfEntry && onPerfEntry instanceof Function) {
+    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
```
что ты тут сделал? ты чё?

EcLiPsE-code committed:
add react-redux

+ create gloabal sore
+ added reducers for main screens
+ added action creators

EcLiPsE-code committed:
was add react-redux on page with tests

EcLiPsE-code committed:
update redux part in test page

EcLiPsE-code committed:
bug fixed

+ added redux on something pages
+ added authorization
